src/Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp

```diff
@@ -7,6 +7,7 @@
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Particles/MonteCarlo/NeutrinoInteractionTable.hpp"
```

Comment:
Anything with a `std::` in this file needs the appropriate include.  All of the spectre includes should just be forward declarations and an include of `DataStructures/Tensor/TypeAliases.hpp`.  You may need to add these includes to the cpp file.